### PR TITLE
Add interpolations for MIRK methods

### LINKED ISSUE #111: Interoplation for MIRK methods
We need to implement interpolations for MIRK methods so that we can approximate the solution between mesh points.

### PR DESCRIPTION
**Pull request by ErikQQY**
Fix #111

This PR adds interpolations for MIRK methods

Tested on example problem from README:

image

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Can you add a convergence with something like a mid time condition in order test the correctness here?

**Comment by avik-pal** [comment]
This only adds interpolation to the solution object, right? We need the intermediate objects passed to the `bc` to be interpolation-friendly as well.

**Comment by ChrisRackauckas** [comment]
Yes, that `sol` object is what you should just be passing to the bc anyways, just then updating its coefficient vectors as you go along.

**Comment by avik-pal** [comment]
This would also be a breaking change, right? We can up the SciMLBase compat to 2.0 and do that together with https://github.com/SciML/BoundaryValueDiffEq.jl/pull/109 & https://github.com/SciML/SciMLBase.jl/pull/477

**Comment by ChrisRackauckas** [comment]
Why would this be a breaking change?

**Comment by avik-pal** [comment]
That changes the input specification to `bc`. The current specification is a vector of arrays which now changes to a solution similar to shooting.

**Comment by ChrisRackauckas** [comment]
It was supposed to always be the solution. The solution type acts as a vector of arrays so the actions are the same, just extended though?

File: src/adaptivity.jl [hunk 1]
```diff
@@ -253,7 +253,7 @@ end
         w′[(stage + 1):s_star],
         true,
         true)
-    z .= z .* dt .+ cache.y₀[i]
+    z .= z .* dt[1] .+ cache.y₀[i]
```
Comment (ChrisRackauckas) [inline src/adaptivity.jl:256]: this `1` seems odd?

**Pull request merged**